inheritance_polymorphism.rb

```diff
@@ -0,0 +1,39 @@
+
+class Organization
+  attr_accessor :name1, :employees, :location, :founded
+
+  def initialize(name, employees, location, founded)
+    @name1 = name
+    @employees = employees
+    @location = location
+    @founded = founded
+  end  
+  
+  def print_name
+    puts name1
+  end
+end
+
+
+class BusinessUnit < Organization
+  def initialize(name, employees, location, founded, department)
+    super(name, employees, location, founded)
+    @department = department
+  end
+
+  def print_name
+    self.name1 = "BU: #{name1}"
+    puts name1
+  end
```

Comment:
We should not update the value if the method name is print_name. Either update the name or remove the value updation part.